src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.ShoppingCartDao;
+import com.dev.cinema.exeption.DataProcessingExeption;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl extends GenericDaoImpl<ShoppingCart> implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart entity) {
+        return super.add(entity);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM ShoppingCart sc "
+                    + "LEFT JOIN FETCH sc.tickets "
+                    + "LEFT JOIN FETCH sc.user "
+                    + "WHERE sc.user = :user", ShoppingCart.class)
+                    .setParameter("user", user)
+                    .uniqueResult();
+        } catch (HibernateException e) {
+            throw new DataProcessingExeption("Couldn't get shopping cart id=" + user.getId(), e);
```

Comment:
```suggestion
            throw new DataProcessingExeption("Couldn't get shopping cart by user id=" + user.getId(), e);
```
або ж як писала Даша, ми можемо не використовувати блок catch з операціями читання - це опціонально

Reply:
Так, згоен. Але захотілось поставити кетч, чомусь мені так воно симпатичніше :)